### DISCUSSION TITLE
[How to] Draw image to framebuffer in HiDpi

### DISCUSSION THREAD

**Discussion by eric5f3759df** [fltk-rs/fltk-rs#931] [Q&A]
I cannot find a way to draw Image to frame by HiDpi, only can 1 point - 1px.

**Comment by MoAlyousef** [comment]
The crate provides Window::pixels_per_unit(), pixel_w and pixel_h which you can use to draw images. You would probably want to do your own drawing in a widget's draw() method.
```rust
let image =  /* get image data somehow */;
some_widget.draw(|w| {
   draw::draw_image(&image.to_rgb_data(), f.x(), f.y(), image.w(), image.h(), image.depth()).unwrap();
});
```
https://docs.rs/fltk/1.2.7/fltk/draw/fn.draw_image.html
Also see this example here:
https://github.com/parasyte/pixels/blob/master/examples/minimal-fltk/src/main.rs
